### ISSUE TITLE
Not able to use CustomComparator for a object in list

### ISSUE THREAD

**Issue by skarian92** [javers/javers#492]
Hi 

I want to compare two objects(of type Sample2) which have one list member of a custom object(of type sample). I wrote custom comparator for Sample class but javers code never hits the custom comparator.

For eg:

<pre>public class Sample {

  private String one;

  private final int two;

  public Sample(String one, int two) {
    this.one = one;
    this.two = two;
  }
}
</pre>

<pre>public class Sample2 {
  String id;
  
private List<Sample> sample;

  public Sample2(String id, List<Sample> sample) {
    this.id = id;
    this.sample = sample;
  }
}</pre>

CustomComparator:
<pre>
public class CustomSampleComparator implements CustomPropertyComparator<Sample, ValueChange> {

  public ValueChange compare(Sample left, Sample right, GlobalId affectedId, Property property) {
    System.out.println("Comparator called...");
    return new ValueChange(affectedId, "testing", 3, 6);
  }

}</pre>

Main class:
<pre>public class App {
  public static void main(String[] args) {

    Sample type1 = new Sample("34", 54);
    Sample type2 = new Sample("55", 89);

    List<Sample> samples1 = new ArrayList<Sample>();
    samples1.add(type1);
    samples1.add(type2);

    Sample type3 = new Sample("34", 4);
    Sample type4 = new Sample("55", 9);

    List<Sample> samples2 = new ArrayList<Sample>();
    samples2.add(type3);
    samples2.add(type4);

    Sample2 sam = new Sample2("1", samples1);
    Sample2 sam2 = new Sample2("1", samples2);

    Javers javers =
        JaversBuilder.javers().registerCustomComparator(new CustomSampleComparator(), Sample.class)
            .build();

    Diff diff = javers.compare(sam, sam2);
    System.out.println(diff.prettyPrint());
    System.out.println("Hello World!");
  }
}
</pre>

**Problem**: My custom comparator is never bieng hit.

**Comment by bartoszwalacik** [comment]
For now, CustomPropertyComparator compares only property values and not items inside collections. We need to check if it possible to use it as a collections Comparator.

**Comment by bartoszwalacik** [comment]
fixed in release 3.3.0, see https://javers.org/release-notes/

**Comment by bartoszwalacik** [comment]
for example :
```java
 javersBuilder.registerValue(BigDecimal.class, (a,b) -> a.compareTo(b) == 0);
```
